src/App.tsx

```diff
@@ -15,71 +16,107 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+interface States {
+  selectedGood: string;
+}
+
+export class App extends Component<{}, States> {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  select = (good: string) => {
+    if (good === this.state.selectedGood) {
+      this.setState({
+        selectedGood: '',
+      });
```

Comment:
don't repeat yourself: let's check once is current good selected, according to those check will define the needed value that we need to set to the state, and call `setState` only once

Reply:
I didn't completely understand what you meant, but I did some changes, please check

Reply:
will be better to create some variable, like `newGood`, in which you'll store empty string or `good` value according to the check: is current good selected
and then only one time pass this variable to setState method, instead of copying your code and calling setState twice in one function